src/native/managed/cdacreader/src/Target.cs

```diff
@@ -138,6 +157,29 @@ private Target(Configuration config, ContractDescriptorParser.ContractDescriptor
         return true;
     }
 
+    public byte ReadUInt8(ulong address)
```

Comment:
For these `ReadX` and `TryReadX` methods, what about abstracting it in terms of `T` where `T : struct, IBinaryInteger<T>`? Then you could use `IBinaryInteger<T>.Try(Read/Write)(Little/Big)Endian`.

Floating point support can be done manually on top of that (mapping to and from integer types of the correct size).